### PR TITLE
Add support for Firefox to extension

### PR DESCRIPTION
**Pull request by lucacasonato**
Adding support for Firefox to the extension is relatively simple because both Chrome and Firefox use the WebExtensions API. In our case this means that the same code will work as long as you place the native host manifests into the correct directory for Firefox. I added this to the extension setup command.

Annoyingly Firefox handles its Content Security Policy differently to Chrome for extensions (https://bugzilla.mozilla.org/show_bug.cgi?id=1267027). In a nutshell this means that we can not call `ws://127.0.0.1:XXXXX/api/v1/run` from the content script itself. This means that the ws needs to be opened in the background script, which can then proxy messages to the content script. I have implemented this too.

I also changed the command to setup the native host manifests to `sail setup-extension` (because it does both Chrome and Firefox).

### CODE REVIEW

**Comment by lucacasonato** [comment]
There is no .prettierrc file or similar included, so I formatted it in a way that seemed to fit most. If there is a .prettierrc file anywhere that belongs to this it would be nice if this was added to the repo.

**Comment by deansheather** [comment]
I think the old command `install-for-chrome-ext` should be aliased to the new one, even if it gets a deprecated warrning.

@kylecarbs Could you please review this?

**Comment by lucacasonato** [comment]
@kylecarbs

File: extension/src/content.ts [hunk 1]
```diff
@@ -1,40 +1,43 @@
...
-					onMessage({ type, v: content });
-					break;
-				default:
-					throw new Error("unknown message type: " + type);
-			}
-		});
-	});
+}) => void): Promise<chrome.runtime.Port> => {
+  return new Promise<chrome.runtime.Port>((resolve, reject) => {
+    const port = chrome.runtime.connect();
```
Comment (kylecarbs) [inline extension/src/content.ts]: Do `chrome.*` API calls still work in firefox?

Comment (lucacasonato) [reply extension/src/content.ts]: Yes. 

> Firefox also implements these APIs under the chrome namespace using callbacks. This allows code written for Chrome to run largely unchanged in Firefox for the APIs documented here. 

https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API#wikiArticle

**Comment by deansheather** [comment]
Working on this in a new PR.

**Pull request closed without merging**